maliput/src/test_utilities/CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 ##############################################################################
 # Sources
 ##############################################################################
+if(BUILD_TESTING)
+
```

Comment:
Not fully getting why it is needed. Have you added this because of the dependency of a couple of the files to gtest?

Reply:
I see, it is related to this issue https://github.com/maliput/maliput/issues/501

Reply:
Without this change the build will fail on the buildfarm because it will only get the declared dependencies not the test dependencies. The alternative is to add gtest as a core dependecy which isn't great. 

Example failure https://build.ros2.org/job/Fpr__maliput__ubuntu_focal_amd64/1/console

```14:41:42 --- stderr: maliput
14:41:42 In file included from [01m[K/tmp/ws/src/maliput/maliput/src/test_utilities/check_id_indexing.cc:30[m[K:
14:41:42 [01m[K/tmp/ws/src/maliput/maliput/include/maliput/test_utilities/check_id_indexing.h:32:10:[m[K [01;31m[Kfatal error: [m[Kgtest/gtest.h: No such file or directory
14:41:42    32 | #include [01;31m[K<gtest/gtest.h>[m[K
14:41:42       |          [01;31m[K^~~~~~~~~~~~~~~[m[K
14:41:42 compilation terminated.
14:41:42 make[2]: *** [src/test_utilities/CMakeFiles/test_utilities.dir/build.make:63: src/test_utilities/CMakeFiles/test_utilities.dir/check_id_indexing.cc.o] Error 1
14:41:42 make[1]: *** [CMakeFiles/Makefile2:763: src/test_utilities/CMakeFiles/test_utilities.dir/all] Error 2
14:41:42 make: *** [Makefile:141: all] Error 2
14:41:42 ---
14:41:42 Failed   <<< maliput [1min 47s, exited with code 2]
14:41:42 
14:41:42 Summary: 0 packages finished [1min 47s]
14:41:42   1 package failed: maliput
```